temporal-sdk/src/main/java/io/temporal/client/WorkflowInvocationHandler.java

```diff
@@ -394,4 +399,65 @@ public <R> R getResult(Class<R> resultClass) {
       throw new IllegalStateException("No result is expected");
     }
   }
+
+  private static class UpdateWithStartInvocationHandler implements SpecificInvocationHandler {
+
+    private int step;
```

Comment:
What is the purpose of this variable? I think it is trying to create a little state machine, but it isn't super obvious to me.

Reply:
`invoke` is called twice: once for update and once for start. This keeps track of which one we're expecting. I can rename it to `invocation`?

Reply:
Since each invocation is different can we declare an enum inside this class and use that to keep track of the state?

```java
if(state.equals(NOT_STARTED)) {
       UpdateMethod updateMethod = method.getAnnotation(UpdateMethod.class);
        if (updateMethod == null) {
          throw new IllegalArgumentException(
              "Method '" + method.getName() + "' is not an UpdateMethod");
        }
        this.operation.prepareUpdate(
            untyped,
            methodMetadata.getName(),
            method.getReturnType(),
            method.getGenericReturnType(),
            args);
         state = UPDATE_RECIEVED
        return;
} else if (state.equals(UPDATE_RECIEVED)) {

        WorkflowMethod workflowMethod = method.getAnnotation(WorkflowMethod.class);
        if (workflowMethod == null) {
          throw new IllegalArgumentException(
              "Method '" + method.getName() + "' is not a WorkflowMethod");
        }
        this.operation.prepareStart(untyped);
        state = WORKFLOW_RECIEVED
        return;
} else {
      throw new IllegalArgumentException("UpdateWithStartInvocationHandler called too many times");
}
```

What do you think?

Reply:
That looks better; done! https://github.com/temporalio/sdk-java/pull/2199/commits/7626657a7d96780e6f77930e7f850cfdfe098bd1